corporate/lib/decorator.py

```diff
@@ -57,3 +60,30 @@ def _wrapped_view_func(
         return view_func(request, remote_realm, *args, **kwargs)
 
     return _wrapped_view_func
+
+
+def authenticated_remote_server_management_endpoint(
+    view_func: Callable[Concatenate[HttpRequest, RemoteZulipServer, ParamT], HttpResponse]
+) -> Callable[Concatenate[HttpRequest, ParamT], HttpResponse]:  # nocoverage
```

Comment:
I added this thinking I would need it but looks like `remote_server_path` just gives us the correct `RemoteZulipServer` so didn't need to use it in this PR.

Reply:
Hmm I think this is still likely to be helpful, since we won't do the `RemoteZulipServer` flow for legacy server via `remote_server_path`, but through a login form - https://github.com/zulip/zulip/pull/27970

Reply:
We can't use `remote_server_path` for this part of the API since it's got a browser session, not the org_id/org_key credentials that the server has; so we indeed need this function.